libraries/chain/controller.cpp

```diff
@@ -1940,24 +1938,26 @@ struct controller_impl {
          EOS_ASSERT( std::holds_alternative<completed_block>(pending->_block_stage), block_validate_exception,
                      "cannot call commit_block until pending block is completed" );
 
-         auto bsp = std::get<completed_block>(pending->_block_stage)._block_state;
+         const auto& bsp = std::get<completed_block>(pending->_block_stage)._block_state;
 
-         if( add_to_fork_db ) {
+         if( s == controller::block_status::incomplete ) {
             fork_db.add( bsp );
             fork_db.mark_valid( bsp );
             emit( self.accepted_block_header, bsp );
-            head = fork_db.head();
-            EOS_ASSERT( bsp == head, fork_database_exception, "committed block did not become the new head in fork database");
+            EOS_ASSERT( bsp == fork_db.head(), fork_database_exception, "committed block did not become the new head in fork database");
+         } else if (s != controller::block_status::irreversible) {
+            fork_db.mark_valid( bsp );
          }
+         head = bsp;
 
          // at block level, no transaction specific logging is possible
-         if (auto dm_logger = get_deep_mind_logger(false)) {
+         if (auto* dm_logger = get_deep_mind_logger(false)) {
```

Comment:
I don't see the benefit of adding the `*` here? I could see it if it was `const auto*` maybe. What am I missing?

Reply:
`on_accepted_block()` is not `const` although it should be. the `*` was a clang tidy popup recommendation that I just accepted. I think it likes the `auto*` over `auto` as it makes it more explicit that it is a pointer.